bookstore/handlers.py

```diff
@@ -18,6 +24,73 @@ class BookstoreVersionHandler(APIHandler):
     def get(self):
         self.finish(json.dumps({"bookstore": True, "version": version}))
 
+# NOTE: We need to ensure that publishing is not configured if bookstore settings are not
+# set. Because of how the APIHandlers cannot be configurable, all we can do is reach into settings
+
+
+class BookstorePublishHandler(APIHandler):
+    """Publish a notebook to the publish path"""
+
+    def __init__(self, *args, **kwargs):
+        super(APIHandler, self).__init__(*args, **kwargs)
+        # create an easy helper to get at our bookstore settings quickly
+        self.bookstore_settings = BookstoreSettings(config=self.settings['config']['BookstoreSettings'])
+
+        self.fs = s3fs.S3FileSystem(
+            key=self.bookstore_settings.s3_access_key_id,
+            secret=self.bookstore_settings.s3_secret_access_key,
+            client_kwargs={
+                "endpoint_url": self.bookstore_settings.s3_endpoint_url,
+                "region_name": self.bookstore_settings.s3_region_name,
+            },
+            config_kwargs={},
+            s3_additional_kwargs={},
+        )
+
+    @property
+    def delimiter(self):
+        """It's a slash! Normally this could be configurable. This leaves room for that later,
+        keeping it centralized for now"""
+        return "/"
+
+    @property
+    def full_prefix(self):
+        """Full prefix: bucket + published prefix"""
+        return self.delimiter.join(
+            [self.bookstore_settings.s3_bucket, self.bookstore_settings.published_prefix]
+        )
+
+    def s3_path(self, path):
+        """compute the s3 path based on the bucket, prefix, and the path to the notebook"""
+        return self.delimiter.join([self.full_prefix, path])
+
+    def _publish(self, model, path):
+        if model['type'] != 'notebook':
+            raise web.HTTPError(400, "bookstore only publishes notebooks")
+        content = model['content']
```

Comment:
Is anything happening with the content here? It looks like it's being set to this local variable but then nothing is actually then using that local variable.

Reply:
That's correct, no publish has happened here. I've only hooked up the config and set up the handler. I'm putting integration tests in place first then implementing the actual publish to S3.